### DISCUSSION TITLE
server keeps crashing

### DISCUSSION THREAD

**Discussion by gurinina** [ropensci/webchem#419] [General]
i am usiing cir_img from the r package webchem and the server keeps quitting on me, is there a better alternative? i am simply trying to convert a list of pubchem IDs into png files of structurel

also, the cir_img. doesn't take pubchem IDs as inputs

to reproduce the problem. simply  input 100 or so query's and write them in a for loop:

I get the service is not available

```r
for(i in 1:length(cmp)) cir_img(query = cmp[i],dir = felke,width=300,height = 300,linewidth = 3,symbolfontsize = 12)
```

**Comment by stitam** [comment]
Thanks @gurinina for starting this discussion.

Can you please check that CIR is still not working? This example works on my end: 

``` r
webchem::cir_img("CCO", dir = tempdir())
```
<sup>Created on 2024-07-15 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

I don't think `cir_img()` works with PubChem CIDs, you should convert them to a representation that is not a database id, e.g. smiles:

``` r
# get smiles for pubchem cids 176 and 702
id <- webchem::pc_sect(c("176", "702"), "canonical smiles")
# download images for both
webchem::cir_img(id$Result, dir = tempdir())
```
<sup>Created on 2024-07-15 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

So, I would query your compounds like this:

``` r
library(webchem)
cmp_smiles <- pc_sect(cmp, "canonical smiles")
cir_img(
  query = cmp_smiles$Result, 
  dir = felke,
  width=300,
  height = 300,
  linewidth = 3,
  symbolfontsize = 12
)
```

Does this work?
